exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/metadata/FileMetadataCollector.java

```diff
@@ -240,20 +240,23 @@ private void addColumnMetadata(String[] columnName,
   private static class ColTypeInfo {
 
     OriginalType originalType;
+    List<OriginalType> parentTypes;
     int precision;
     int scale;
     int repetitionLevel;
     int definitionLevel;
 
-    ColTypeInfo(OriginalType originalType, int precision, int scale, int repetitionLevel, int definitionLevel) {
+    ColTypeInfo(OriginalType originalType, List<OriginalType> parentTypes,
+                int precision, int scale, int repetitionLevel, int definitionLevel) {
       this.originalType = originalType;
+      this.parentTypes = parentTypes;
       this.precision = precision;
       this.scale = scale;
       this.repetitionLevel = repetitionLevel;
       this.definitionLevel = definitionLevel;
     }
 
-    static ColTypeInfo of(MessageType schema, Type type, String[] path, int depth) {
+    static ColTypeInfo of(MessageType schema, Type type, String[] path, int depth, List<OriginalType> originalTypes) {
```

Comment:
Please use consistent naming. Field and constructor param was named ```parentTypes``` but here  ```originalTypes``` .